lib/publiccloud/instance.pm

```diff
@@ -165,6 +165,12 @@ sub ssh_script_run {
     my $self = shift;
     my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
     my $ssh_cmd = $self->_prepare_ssh_cmd(%args);
+    $args{timeout} //= SSH_TIMEOUT;
+    if(exists $args{proceed_on_failure} and $args{proceed_on_failure} == 1) {
+        my $external_timeout = $args{timeout};
+        $args{timeout}=$args{timeout}+20;
+        $ssh_cmd = "timeout --foreground -k 10s $external_timeout " . $ssh_cmd;
+    }
```

Comment:
Sharing my experience from another PR, the command `timeout $t $cmd1` can be very critical and leading to faults when using $cmd1 = `sudo $cmd2`.
In such cases, `sudo timeout ... ` should be recommended, moving sudo on top.

Reply:
can you show certain examples ? I don't see how permission of user who executing ssh call can be related to permission of user on remote system this does not make sense to me

Reply:
`timeout ssh sut longcommand` vs `sudo timeout ssh sut longcommand` makes no difference to me.